trace_decoder/src/lib.rs

```diff
@@ -1,92 +1,63 @@
-//! <div class="warning">
-//! This library is undergoing major refactoring as part of (#275)(https://github.com/0xPolygonZero/zk_evm/issues/275).
-//! Consider all TODOs to be tracked under that issue.
-//! </div>
+//! An _Ethereum Node_[^1] executes _transactions_ in _blocks_.
 //!
-//! Your neighborhood zk-ready [ethereum](https://github.com/0xPolygonZero/erigon)
-//! [node](https://github.com/0xPolygonHermez/cdk-erigon/) emits binary "witnesses"[^1].
+//! Execution mutates two key data structures:
+//! - [The state trie](https://ethereum.org/en/developers/docs/data-structures-and-encoding/patricia-merkle-trie/#state-trie).
+//! - [The storage tries](https://ethereum.org/en/developers/docs/data-structures-and-encoding/patricia-merkle-trie/#storage-trie).
 //!
-//! But [`plonky2`], your prover, wants [`GenerationInputs`].
+//! Ethereum nodes expose information about the transactions over RPC, e.g:
+//! - [The specific changes to the storage tries](TxnTrace::storage_written).
+//! - [Changes to account balance in the state trie](TxnTrace::balance).
 //!
-//! This library helps you get there.
+//! The tries and the transactions can be proved to be correct using a
+//! zero-knowledge prover like [`evm_arithmetization`],
+//! an Ethereum-y frontend for [`plonky2`].
 //!
-//! [^1]: A witness is an attestation of the state of the world, which can be
-//!       proven by a prover.
+//! **Prover perfomance is a high priority.**
 //!
-//! # Non-Goals
-//! - Performance - this won't be the bottleneck in any proving system.
-//! - Robustness - malicious or malformed input may crash this library.
+//! The aformentioned trie structures may have subtries _deferred_.
```

Comment:
```suggestion
//! The aforementioned trie structures may have subtries _deferred_.
```

Reply:
deferred sounds off to me (it's used several times across this PR), when hashing out a subtrie we don't defer anything (i.e. execution for instance), we just _obfuscate_ information

Reply:
I've replaced this with "indirect"/"indirected" - how does that sound?

Reply:
Why not the (I think formerly used) `hashed out`?
When I read this:

> //! The principle concern of this library is to step through the transactions,
> //! and reproduce the _intermediate tries_,
> //! while indirecting all possible subtries to minimise prover load
> //! (since prover performance is sensitive to the size of the trie).

I honestly have no clue what _indirecting subtries_ means here, and even with this discussion in mind I find it fairly hard to map to what it does.

Reply:
_hash out_ is honestly my first choice, but it's overloaded in our domain* and I don't want to reuse it. i.e I've seen `type Hashout = ...` somewhere in `smt_trie` I think, and `struct Hashout` in `evm_arith` iirc. If we can't find a better term maybe it's worth the confusion 

\* c.f DDD

Reply:
`To hash something out` is litterally what it is though, we compress the information through the hash function to get a short, random digest.
The `HashOut` types in `plonky2` / `evm_arithmetization` are a shortcut for `Hash Output`. I don't find the use of both cases in the same workspace confusing, but if you do, then I'd suggest that we find something that suits everyone :D